geopandas/io/arrow.py

```diff
@@ -725,5 +781,114 @@ def _read_feather(path, columns=None, **kwargs):
+    _validate_metadata(table.schema.metadata)
     return _arrow_to_geopandas(table)
+
+
+def _read_parquet_schema_and_metadata(path, filesystem):
+    ds = import_optional_dependency(
+        "pyarrow.dataset", extra="pyarrow is required for Feather support."
+    )
+    parquet = import_optional_dependency(
+        "pyarrow.parquet", extra="pyarrow is required for Parquet support."
+    )
+    try:
+        schema = ds.dataset(path, filesystem=filesystem).schema
+    except Exception:
+        schema = parquet.read_schema(path, filesystem=filesystem)
+
+    metadata = schema.metadata
+
+    # if GDAL-produced malformed file
+    if metadata is None or b"geo" not in metadata:
+        try:
+            # read_metadata does not accept a filesystem keyword, so need to
+            # handle this manually (https://issues.apache.org/jira/browse/ARROW-16719)
+            if filesystem is not None:
+                pa_filesystem = _ensure_arrow_fs(filesystem)
+                with pa_filesystem.open_input_file(path) as source:
+                    metadata = parquet.read_metadata(source).metadata
+            else:
+                metadata = parquet.read_metadata(path).metadata
+        except Exception:
+            pass
+
+    return schema, metadata
+
+
+def _validate_metadata(metadata):
+    if metadata is None or b"geo" not in metadata:
+        raise ValueError(
+            """Missing geo metadata in Parquet/Feather file.
+            Use pandas.read_parquet/read_feather() instead."""
+        )
+
+    # check for malformed metadata
+    try:
+        decoded_geo_metadata = _decode_metadata(metadata.get(b"geo", b""))
+    except (TypeError, json.decoder.JSONDecodeError):
+        raise ValueError("Missing or malformed geo metadata in Parquet/Feather file")
+
+    _validate_geo_metadata(decoded_geo_metadata)
+    return decoded_geo_metadata
+
+
+def _get_parquet_bbox_filter(geo_metadata, bbox):
+
+    _validate_bbox_column_in_parquet(geo_metadata)
+
+    bbox_column_name = _get_bbox_encoding_column_name(geo_metadata)
+    return _convert_bbox_to_parquet_filter(bbox, bbox_column_name)
+
+
+def _convert_bbox_to_parquet_filter(bbox, bbox_column_name):
+    import pyarrow.compute as pc
+
+    return ~(
+        (pc.field((bbox_column_name, "xmin")) > bbox[2])
+        | (pc.field((bbox_column_name, "ymin")) > bbox[3])
+        | (pc.field((bbox_column_name, "xmax")) < bbox[0])
+        | (pc.field((bbox_column_name, "ymax")) < bbox[1])
+    )
+
+
+def _validate_bbox_column_in_parquet(geo_metadata):
+    if not _check_if_bbox_column_in_parquet(geo_metadata):
+        raise ValueError("No covering bbox in parquet file.")
+
+    for var in ["xmin", "ymin", "xmax", "ymax"]:
+        if var not in geo_metadata["columns"]["geometry"]["covering"]["bbox"].keys():
+            raise ValueError("Metadata for bbox column is malformed.")
+
+    return True
+
+
+def _check_if_bbox_column_in_parquet(geo_metadata):
```

Comment:
seems like this does not have the most descriptive name, since this appears to check if the "covering" metadata key is present?